### ISSUE TITLE
[MASK] token representation

### ISSUE THREAD

**Issue by g-luo** [openai/CLIP#14]
Hello! I was wondering what is considered a "[MASK]" token in byte pair encoding / the tokenizer.py for CLIP / if there is a standard method for marking masked words?

**Comment by jongwook** [comment]
Our tokenizer only has `<|startoftext|>` and `<|endoftext|>` as special tokens, and no `[MASK]` because CLIP is not a masked language model. We use [a triangular attention mask](https://github.com/openai/CLIP/blob/main/model.py#L292-L298) to have an option to transfer from/to autoregressive language models. You can edit [simple_tokenizer.py](https://github.com/openai/CLIP/blob/main/simple_tokenizer.py) to add support, if you'd like to fine-tune the model using a mask-based method.

Alternatively, in case you just need a tokenizer, you can choose from [existing tokenizers](https://huggingface.co/models) that supports `[MASK]`, or build your own using [Huggingface Tokenizers](https://github.com/huggingface/tokenizers)

**Comment by g-luo** [comment]
Thanks!

**Comment by g-luo** [comment]
Hello, I was wondering what the distribution of words looks like for the captions CLIP has trained on? Ie is there some kind of neutral word that could possibly function like a "[MASK]" token given the dataset CLIP was trained on?

**Comment by Newmu** [comment]
CLIP is trained with all lower case but I believe has uppercase tokens as part of the first 256 "byte" vocab slots. You could probably use one of those untrained uppercase letters as a mask token if you'd like.
